Code/GraphMol/MolHash/hashfunctions.cpp

```diff
@@ -642,23 +690,22 @@ std::string TautomerHashv2(RWMol *mol, bool proto, bool useCXSmiles,
           std::cerr << "  check neighbor " << nbrBnd->getIdx() << " from "
                     << atm->getIdx() << "-" << oatom->getIdx() << std::endl;
           std::cerr << "    " << bondsConsidered[nbrBnd->getIdx()] << " icao "
-                    << isCandidateAtom(oatom) << " hsbo "
+                    << isCandidateAtom(oatom, atomFlags) << " hsbo "
                     << hasStartBond(oatom, startBonds) << " unsat "
                     << isUnsaturatedBond(nbrBnd) << " icaa "
-                    << isCandidateAtom(atm) << " hsba "
-                    << hasStartBond(atm, startBonds) << std::endl;
+                    << isCandidateAtom(atm, atomFlags) << " hsba "
+                    << hasStartBond(atm, startBonds)
+                    << " ncno: " << numConjugatedNeighbors[oatom->getIdx()]
+                    << std::endl;
 #endif
-          if (bondsConsidered[nbrBnd->getIdx()] ||
-              (skipNeighborBond(atm, oatom, nbrBnd, startBonds, atomFlags,
+          if (checkForOverreach(atm, oatom, bnd, nbrBnd, startBonds,
```

Comment:
The logic to set up the `bq` and the logic within the bfs is really similar. But it's not _exactly_ the same. Why is the first bond treated differently than subsequent bonds? I can't quite understand. 

The difference makes me worry that future or present safety checks like this will miss one of the two branches. (I'm fine if the answer is "it should be, that's a separate refactor)

Reply:
I think it's worth considering a future refactor where we try and make it a single loop.
When I originally wrote this code that didn't seem doable, but there have been enough changes now that it may work.

Reply:
that makes sense. I agree it's out of scope for this PR - I brought it up because it was a barrier to my understanding.

thanks